src/openldap/mod.rs

```diff
@@ -0,0 +1,519 @@
+        self
+    }
+
+    /// Sets an accesslog admin user up (there can only be one)
+    /// Configuring the admin for the access log can be done via [`OpenLDAP::with_accesslog_admin`]
+    /// Default: `None`
+    pub fn with_accesslog_settings(
+        mut self,
+        AccesslogSettings {
+            log_operations,
+            log_success,
+            log_purge,
+            log_old,
+            log_old_attribute,
+        }: AccesslogSettings,
+    ) -> Self {
+        self.env_vars
+            .insert("LDAP_ENABLE_ACCESSLOG".to_owned(), "yes".to_owned());
+        let log_operations = match log_operations {
+            AccesslogLogOperations::Writes => "writes".to_owned(),
+            AccesslogLogOperations::Reads => "Reads".to_owned(),
+            AccesslogLogOperations::Session => "Session".to_owned(),
+            AccesslogLogOperations::All => "All".to_owned(),
+        };
+        self.env_vars
+            .insert("LDAP_ACCESSLOG_LOGOPS".to_owned(), log_operations);
+        if log_success {
+            self.env_vars
+                .insert("LDAP_ACCESSLOG_LOGSUCCESS".to_owned(), "TRUE".to_owned());
+        } else {
+            self.env_vars
+                .insert("LDAP_ACCESSLOG_LOGSUCCESS".to_owned(), "FALSE".to_owned());
+        }
+        self.env_vars.insert(
+            "LDAP_ACCESSLOG_LOGPURGE".to_owned(),
+            format!("{} {}", log_purge.0, log_purge.1),
+        );
+        self.env_vars
+            .insert("LDAP_ACCESSLOG_LOGOLD".to_owned(), log_old.to_owned());
+        self.env_vars.insert(
+            "LDAP_ACCESSLOG_LOGOLDATTR".to_owned(),
+            log_old_attribute.to_owned(),
+        );
+        self
+    }
+
+    /// Activates the access log and sets the admin user up (there can only be one)
+    /// Configuring how [`OpenLDAP`] logs can be done via [`OpenLDAP::with_accesslog_settings`]
+    /// Default: `None`
+    pub fn with_accesslog_admin(mut self, username: &str, password: &str) -> Self {
+        self.env_vars
+            .insert("LDAP_ENABLE_ACCESSLOG".to_owned(), "yes".to_owned());
+        self.env_vars.insert(
+            "LDAP_ACCESSLOG_ADMIN_USERNAME".to_owned(),
+            username.to_owned(),
+        );
+        self.env_vars.insert(
+            "LDAP_ACCESSLOG_ADMIN_PASSWORD".to_owned(),
+            password.to_owned(),
+        );
+        self
+    }
+
+    /// Adds a user (can be called multiple times)
+    /// Default: `[]`
+    ///
+    /// Alternatively, users can be added via [`OpenLDAP::with_users`].
+    pub fn with_user(mut self, username: &str, password: &str) -> Self {
+        self.users.push(User {
+            username: username.to_owned(),
+            password: password.to_owned(),
+        });
+        self
+    }
+
+    /// Add users (can be called multiple times)
+    /// Default: `[]`
+    ///
+    /// Alternatively, users can be added via [`OpenLDAP::with_user`].
+    pub fn with_users<Username: AsRef<str>, Password: AsRef<str>>(
```

Comment:
We actually convert it to `String` anyway. So, I’d prefer to expect `impl Into<String>` thus we can avoid extra allocation if `String` has been passed to functions. And still works for `&str`.

Could you re-check all functions?
Thank you 🙏

Reply:
good call, have changed as such. Makes the code even a bit tidier (even though that part is subjective ^^)

Reply:
Okay, This is hitting my rust-knowlege-limits.

I need an `[String]`, to push into the vector
=> the best I can make the borrow-checker happy with is `<S: Into<String> + Clone>`.. Seems very unergonomic..

Should I switch back to just using `String`?

Reply:
I've left a comment here https://github.com/testcontainers/testcontainers-rs-modules-community/pull/180#discussion_r1723219292

]ToString] should be a good fit here

Reply:
`IntoIterator` is much nicer to read. That was the idea that I was missing/not seeing 👍🏻